pkg/server/backuprestoreserver.go

```diff
@@ -151,19 +153,47 @@ func (b *BackupRestoreServer) runServerWithSnapshotter(ctx context.Context, rest
 	handler := b.startHTTPServer(etcdInitializer, ssr)
 	defer handler.Stop()
 
-	ssrStopCh := make(chan struct{})
+	leaderCallbacks := &leaderelection.LeaderCallbacks{
+		OnStartedLeading: func(ctx context.Context) {
+			// Get the updated snapshotter state
+			err := ssr.GetLatestSnapshotterState()
+			if err != nil {
+				fmt.Printf("Unable to get Latest SnapshotterState: %v", err)
```

Comment:
In this case, we have a leader election and snapshotter is not updated with latest state. The next time it will be rectified is when the leader election changes next time. What happens in the meantime?